spec/requests/follow_relationships_spec.rb

```diff
@@ -0,0 +1,194 @@
+
+      context 'when the request format is HTML' do
+        it 'creates a follow relationship' do
+          expect do
+            post follow_relationships_path, params: { followed_id: followed_user.id }
+          end.to change(FollowRelationship, :count).by(1)
+        end
+
+        it 'creates a following for the follower user' do
+          expect do
+            post follow_relationships_path, params: { followed_id: followed_user.id }
+          end.to change { follower_user.following.count }.by(1)
+        end
+
+        it 'creates a follower for the followed user' do
+          expect do
+            post follow_relationships_path, params: { followed_id: followed_user.id }
+          end.to change { followed_user.followers.count }.by(1)
+        end
+
+        it 'returns 302 Found' do
+          post follow_relationships_path, params: { followed_id: followed_user.id }
+          expect(response).to have_http_status(:found)
+        end
+
+        it 'redirects to the followed user profile' do
+          post follow_relationships_path, params: { followed_id: followed_user.id }
+          should redirect_to(user_profile_path(followed_user.nickname))
+        end
+
+        it 'returns an HTML response' do
+          post follow_relationships_path, params: { followed_id: followed_user.id }
+          expect(response.content_type).to eq 'text/html; charset=utf-8'
+        end
+      end
+
+      context 'when the request format is JavaScript' do
+        it 'creates a follow relationship' do
+          expect do
+            post follow_relationships_path, params: { followed_id: followed_user.id }, xhr: true
+          end.to change(FollowRelationship, :count).by(1)
+        end
+
+        it 'creates a following for the follower user' do
+          expect do
+            post follow_relationships_path, params: { followed_id: followed_user.id }, xhr: true
+          end.to change { follower_user.following.count }.by(1)
+        end
+
+        it 'creates a follower for the followed user' do
+          expect do
+            post follow_relationships_path, params: { followed_id: followed_user.id }, xhr: true
+          end.to change { followed_user.followers.count }.by(1)
+        end
+
+        it 'responds with success' do
+          post follow_relationships_path, params: { followed_id: followed_user.id }, xhr: true
+          expect(response).to have_http_status(:success)
+        end
+
+        it 'returns a JavaScript response' do
+          post follow_relationships_path, params: { followed_id: followed_user.id }, xhr: true
+          expect(response.content_type).to eq('text/javascript; charset=utf-8')
+        end
+
+        it 'changes the follow button to the unfollow button' do
+          post follow_relationships_path, params: { followed_id: followed_user.id }, xhr: true
+          expect(response.body).to include('Unfollow')
+        end
+      end
+    end
+  end
+
+  describe 'DELETE /destroy' do
```

Comment:
Let's add one more test here, to check what happens if you try to unfollow a user that you are not following (do this after you handle this in my previous comment)